components/server/src/api/envvar-service-api.ts

```diff
@@ -0,0 +1,243 @@
+
+        const userEnvVars = await this.envVarService.listUserEnvVars(context.user.id, context.user.id);
+        const userEnvVarfound = userEnvVars.find((i) => i.id === req.envVarId);
+        if (userEnvVarfound) {
+            const variable: UserEnvVarValue = {
+                name: req.name ?? userEnvVarfound.name,
+                value: req.value ?? userEnvVarfound.value,
+                repositoryPattern: req.repositoryPattern ?? userEnvVarfound.repositoryPattern,
+            };
+            variable.repositoryPattern = UserEnvVar.normalizeRepoPattern(variable.repositoryPattern);
+
+            await this.envVarService.updateUserEnvVar(context.user.id, context.user.id, variable);
+
+            const updatedUserEnvVars = await this.envVarService.listUserEnvVars(context.user.id, context.user.id);
+            const updatedUserEnvVar = updatedUserEnvVars.find((i) => i.id === req.envVarId);
+            if (!updatedUserEnvVar) {
+                throw new ConnectError("could not update env variable", Code.Internal);
+            }
+
+            response.environmentVariable = this.apiConverter.toUserEnvironmentVariable(updatedUserEnvVar);
+            return response;
+        }
+
+        throw new ConnectError("env variable not found", Code.InvalidArgument);
+    }
+
+    async createUserEnvironmentVariable(
+        req: CreateUserEnvironmentVariableRequest,
+        context: HandlerContext,
+    ): Promise<CreateUserEnvironmentVariableResponse> {
+        const response = new CreateUserEnvironmentVariableResponse();
+
+        const variable: UserEnvVarValue = {
+            name: req.name,
+            value: req.value,
+            repositoryPattern: req.repositoryPattern,
+        };
+        variable.repositoryPattern = UserEnvVar.normalizeRepoPattern(variable.repositoryPattern);
+
+        await this.envVarService.addUserEnvVar(context.user.id, context.user.id, variable);
+
+        const updatedUserEnvVars = await this.envVarService.listUserEnvVars(context.user.id, context.user.id);
+        const updatedUserEnvVar = updatedUserEnvVars.find(
+            (v) => v.name === variable.name && v.repositoryPattern === variable.repositoryPattern,
+        );
+        if (!updatedUserEnvVar) {
+            throw new ConnectError("could not create env variable", Code.Internal);
+        }
+
+        response.environmentVariable = this.apiConverter.toUserEnvironmentVariable(updatedUserEnvVar);
+
+        return response;
+    }
+
+    async deleteUserEnvironmentVariable(
+        req: DeleteUserEnvironmentVariableRequest,
+        context: HandlerContext,
+    ): Promise<DeleteUserEnvironmentVariableResponse> {
+        const variable: UserEnvVarValue = {
+            id: req.envVarId,
+            name: "",
+            value: "",
+            repositoryPattern: "",
+        };
+
+        await this.envVarService.deleteUserEnvVar(context.user.id, context.user.id, variable);
```

Comment:
why is not enough to pass id?

Reply:
it's needed for old api resource guard validation

Reply:
Already updated a couple 😅